dspace/modules/api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java

```diff
@@ -258,67 +250,77 @@ private String mintAndRegister(Context context, Item item, boolean register) thr
+    // update that previous item to be version 1: dryad.xxxx.1
+    private void mintDOIFirstVersion(Context context, Item item, boolean register) throws SQLException, IOException, AuthorizeException {
         VersionHistory history = retrieveVersionHistory(context, item);
 
         if (history != null) {
             Version version = history.getVersion(item);
             // if it is the first time that is called "create version": mint identifier ".1"
             Version previous = history.getPrevious(version);
+            // if the previous version was the first version,
+            // update that previous item's metadata with the .1 versioned identifier.
             if (history.isFirstVersion(previous)) {
-                String previousDOI= updateIdentifierPreviousItem(context,previous.getItem());
-                DOI firstDOI = new DOI(previousDOI, previous.getItem());
+                Item previousItem = previous.getItem();
+                String previousDOI = getDoiValue(previousItem);
+                String collection = getCollection(context, previousItem);
+                if (collection.equals(myDataFileColl)) {
+                    previousDOI = calculateDOIDataFileFirstTime(previousItem);
+                    String oldDOIstring = getDoiValue(previousItem);
+                    Item dataPackage = DryadWorkflowUtils.getDataPackage(context,previousItem);
+                    if(dataPackage!=null) {
+                        VersionHistory dpHistory = retrieveVersionHistory(context, dataPackage);
+
+                        if (dpHistory != null) {
+                            // if it is the first time that is called "create version": mint identifier ".1"
+                            Version dpHistoryPrevious = dpHistory.getPrevious(dpHistory.getVersion(dataPackage));
+                            if(dpHistoryPrevious!=null && oldDOIstring!=null)
+                            {
+                                Item previousDataPackage = dpHistoryPrevious.getItem();
+
+                                updateHasPartDataFile(context,previousDataPackage,previousDOI,oldDOIstring);
+                                updateIsPartDataFile(context,previousItem,previousDataPackage);
+                            }
+                        }
+                    }
+                } else {
+                    previousDOI = getVersionedDataPackageDOIString(previousDOI,1);
+                }
+                updateItemDOIMetadata(previousItem, previousDOI);
+                DOI firstDOI = new DOI(previousDOI, previousItem);
                 if(DryadDOIRegistrationHelper.isDataPackageInPublicationBlackout(item)) {
-                    mint(firstDOI, "http://datadryad.org/publicationBlackout", register, createListMetadata(previous.getItem()));
+                    mint(firstDOI, "http://datadryad.org/publicationBlackout", register, createListMetadata(previousItem));
```

Comment:
The "http://datadryad.org/publicationBlackout" string seems like it belongs in a properties file.